text/MateuszTokarz/opis.tex

```diff
@@ -0,0 +1,99 @@
+\documentclass[12pt]{article}
+
+\usepackage[utf8]{inputenc}
+\usepackage[T1]{fontenc}
+\usepackage{graphicx}
+\usepackage{xcolor} 
+\usepackage{amsthm}
+\usepackage{amsmath}
+
+\theoremstyle{plain}
+\newtheorem*{tw}{Twierdzenie}
+\newtheorem*{df}{Definicja}
+\newtheorem*{lemat}{Lemat}
+
+\begin{document}
+\title{Aproksymacja najkrótszego nadsłowa}
+\author{Mateusz Tokarz}
+\maketitle
+
+\section*{Opis Problemu}
+Problem najkrótszego nadsłowa w wersji decyzyjnej jest problemem NP zupełnym. W związku z tym pojawiało się wiele podejść do aproksymacji optymalizacyjnej wersji problemu. Tarhio i Ukkonen oraz Turner napisali dwie niezależne prace naukowe, w których opisano algorytmy aproksymacyjne dla rozważanego problemu. Mimo, że autorzy twierdzili, że skonstruowane przez nich algorytmy powinny być $2$-aproksymacyjne, to nie udało im się udowodnić żadnego ograniczenia. 
+
+
+W tym dokumencie przeanalizujemy algorytm opisany przez Minga Li, który jest pierwszą aproksymacyją, dla której udało się udowodnić współczynnik - $log(n)$. Autor pracy twierdzi, że współczynnik w rzeczywistości wynosi co najwyżej $2$.
+  
+\section*{Algorytm}
+Zakładamy, że podane na wejściu słowa spełniają warunek, iż żadne nie jest podsłowem innego. Możemy tak zrobić, ponieważ wyeliminowanie tego typu sytuacji jest niezmiennicze ze względu na wynik i zajmuje wielomianowy czas.
+\newline 
+
+\begin{df}
+Dla danych słów $s_1, s_2$ słowo $m(s_1, s_2)$, długości nie większej niż $|s_1| + |s_2|$, nazywamy scaleniem, jeżeli $s_1$ jest jego prefiksem, a $s_2$ sufiksem (lub $s_2$ prefiksem, a $s_1$ sufiksem).
+\end{df}
+\bigskip
+
+\begin{lemat}
+Dla danych słów $s_1, s_2$ istnieje co najwyżej $2 min(|s_1|, |s_2|)$ scaleń.
+\end{lemat}
+
+Dowód powyższego lematu jest trywialny.
+\newline
+
+\textbf{Algorytm}
+\begin{enumerate}
+    \item Input to $S = \{s_1, \dots, s_n\}$. Jeżeli $|S| = 1$, to zwróć $s_1$.
+    \item Zdefiniujmy $T = \emptyset$.
+    \item Niech $s, s'$ będą słowami minimalizującymi 
+    $$\min_{m - \text{scalenie}} \frac{|m(s, s')|}{v(m(s, s'))},$$
```

Comment:
I guess it'd be less problematic if you'd define e.g. M = {m(s, s'): s, s' \in S} or something like that.
Still more formal it'd be if you define m(s, s', l) and assume that s has to be before s' and length of the overlap is l - then this is a proper function for some subset of (S, S, N).

However, the overall the idea is clear enough, just the formalities are not there.